tests/main/views/test_drafts.py

```diff
@@ -56,6 +62,455 @@ def service_count(self):
+
+    def test_should_400_if_no_questions_to_copy(self):
+        res = self.post_to_copy_published_from_framework(questions_to_copy=None)
+
+        assert res.status_code == 400
+        assert "Required data missing: 'questions_to_copy'" in json.loads(res.get_data(as_text=True))['error']
+
+    def test_should_404_if_framework_does_not_exist(self):
+        res = self.post_to_copy_published_from_framework(framework_slug='z-cloud')
+
+        assert res.status_code == 404
+
+    @pytest.mark.parametrize('framework_status', Framework.STATUSES)
+    def test_should_400_if_target_framework_not_open(self, framework_status):
+        if framework_status == 'open':
+            return
+
+        self.set_framework_status('g-cloud-7', framework_status)
+
+        res = self.post_to_copy_published_from_framework()
+        assert res.status_code == 400
+
+        assert 'Target framework is not open' in res.get_data(as_text=True)
+
+    def test_should_only_copy_published_services_from_correct_framework_and_lot(self):
+        self.setup_dummy_services(3, supplier_id=1, lot_id=1, status='published')
+        self.setup_dummy_services(2, supplier_id=1, lot_id=1, status='disabled', start_id=3)
+        self.setup_dummy_services(4, supplier_id=1, lot_id=2, status='published', start_id=5)
+        self.setup_dummy_services(1, supplier_id=1, lot_id=2, status='disabled', start_id=9)
+        self.setup_dummy_services(2, supplier_id=1, lot_id=3, status='published', start_id=10)
+        self.setup_dummy_services(3, supplier_id=1, lot_id=3, status='enabled', start_id=12)
+
+        services = Service.query.filter(
+            Service.supplier_id == 1,
+            Service.framework_id == 1,  # G-Cloud 6
+            Service.lot_id == 2,
+            Service.status == 'published'
+        ).all()
+
+        res = self.post_to_copy_published_from_framework()
+        assert res.status_code == 201
+
+        drafts = json.loads(res.get_data(as_text=True))['services']
+
+        assert len(drafts) == 4
+        assert all(draft['frameworkSlug'] == 'g-cloud-7' for draft in drafts)
+        assert all(draft['lotSlug'] == 'paas' for draft in drafts)
+        assert all(draft['status'] == 'not-submitted' for draft in drafts)
+        for service, draft in zip(services, drafts):
+            assert service.data['serviceName'] == draft['serviceName']
+
+    def test_should_create_drafts_in_reverse_alphabetical_order_by_service_name(self):
+        for service_id, char in enumerate(('S', 'E', 'R', 'V', 'I', 'C', 'E')):
+            self.setup_dummy_service(
+                str(2000000000 + service_id),
+                lot_id=2,
+                data={'serviceName': f'{char} service'},
+            )
+
+        res = self.post_to_copy_published_from_framework()
+        assert res.status_code == 201
+
+        drafts = json.loads(res.get_data(as_text=True))['services']
+
+        assert sorted(drafts, key=lambda draft: draft['serviceName'], reverse=True) == drafts
```

Comment:
Can we do a comparison against something static here? It's clearer what is happening here and what behaviour we actually want. Eg `assert [draft['serviceName'] for draft in drafts] == ['V service', 'S service', 'R service', 'I service', 'E service', 'E service', 'C service']`